idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -1536,3 +1535,199 @@ def sample_points(self):
+                        len(data_input[0]),
+                    )
+                )
+                bounds_array[0, :] = np.array(data_input[0])
+                bounds_array[1, :] = np.array(data_input[1])
+                data_headers = []
+            self.data = bounds_array
+            self.data_headers = data_headers
+
+            # Catch potential errors in number_of_samples
+            if number_of_samples is None:
+                print(
+                    "\nNo entry for number of samples to be generated. The default value of 5 will be used."
+                )
+                number_of_samples = 5
+            elif not isinstance(number_of_samples, int):
+                raise Exception("number_of_samples must be an integer.")
+            elif number_of_samples <= 0:
+                raise Exception("number_of_samples must a positive, non-zero integer.")
+            self.number_of_samples = number_of_samples
+            self.x_data = bounds_array  # Only x data will be present in this case
+
+        # Check that list_of_distributions is a list, list length is correct and all list values are strings
+        if list_of_distributions is None:
+            raise ValueError("list_of_distributions cannot be empty.")
+        if not isinstance(list_of_distributions, list):
+            raise TypeError("Error with list_of_distributions: list required.")
+        if len(list_of_distributions) != self.x_data.shape[1]:
+            raise ValueError(
+                "Length of list_of_distributions must equal the number of variables."
+            )
+        if all(isinstance(q, str) for q in list_of_distributions) is False:
+            raise TypeError("All values in list must be strings")
+        if not all(
+            q.lower() in ["random", "normal", "uniform"] for q in list_of_distributions
+        ):
+            raise ValueError(
+                "list_of_distributions only supports 'random', 'normal' and 'uniform' sampling options."
+            )
+        self.dist_vector = list_of_distributions
+
+    def generate_from_dist(self, dist_name):
+        if dist_name.lower() in ["uniform", "random"]:
+            dist = getattr(np.random.default_rng(), dist_name.lower())
+            var_values = np.array(dist(size=self.number_of_samples))
+            return dist, var_values
+        elif dist_name.lower() == "normal":
+            dist = getattr(np.random.default_rng(), "normal")
+            var_values = dist(loc=0.5, scale=1 / 6, size=self.number_of_samples)
+            if sum(
+                [1 for i in range(0, var_values.shape[0]) if var_values[i] > 1]
+            ) + sum([1 for i in range(0, var_values.shape[0]) if var_values[i] < 0]):
+                warnings.warn(
+                    "Points adjusted to remain within specified Gaussian bounds."
+                )
+            var_values_truncated = np.array(
+                [1.0 if j > 1.0 else 0.0 if j < 0.0 else j for j in var_values]
```

Comment:
What happens if you have multiple values that exceed the bound? Wouldn't that create the possibility of multiple samples with the same value (even if the chances are very small)?

Reply:
@andrewlee94 yes it would, with enough samples.

The other options are:

1. To return the values as is, but make it clear to the user in the documentation that this is a possibility.
2. Try to re-generate new random numbers from the distribution to replace any values exceeding the bounds. This was my first thought actually, but I'm worried that it could lead to the distribution not being maintained.

So maybe option 1? And make it clear that the bounds are for 99.7%, and that there may be a few values outside of the bounds.

Reply:
I think documenting option 1 is the safest for now - another possibility would be to do 1 with the option of 2 (i.e. give an option to strictly enforce the bounds, noting that this means that the distributions will not be strictly maintained).

Either way, this is hopefully mostly theoretical.

Reply:
Fixed. Doing 1 with the option of 2. Also added tests that check this (by generating generating 10000 samples from the 0-1 distribution)